onadata/apps/api/tasks.py

```diff
@@ -64,6 +64,14 @@ def delete_xform_async(xform_id, user_id):
     xform.soft_delete(user)
 
 
+@app.task()
+def delete_subissions_async(instances, user_id=None):
+    """Soft delete an XForm asynchrounous task"""
+    user = User.objects.get(pk=user_id)
```

Comment:
Seems like the User ID can be none ? Can we skip running this query if it is ?

Reply:
The user should not be nullable

Reply:
thanks for catching this, I'll remove the default value so that we always expect a user ID value